### PR TITLE
[test] fix gcb e2e

### PR DESCRIPTION
**Pull request by ramonpetgrave64**
Followup to https://github.com/slsa-framework/example-package/pull/350
Fixes https://github.com/slsa-framework/slsa-github-generator/issues/3344

This PR fixes the negative test to edit the payload in both of the embedded provenances of the test attestation file.

The GCB attestations are now actually two Provenances in one .jsonl file: one with the 0.1 slsa provenance version, and another with the 1.0 slsa provenance version. The "wrong payload" negative test is meant to edit the Payload of the DSSE Envelope of ONE of the Provenances, so that it should fail signature verification. 

- https://github.com/slsa-framework/example-package/blob/7cd48d610a1184bdd5b80670781d8f4865e86274/.github/workflows/scripts/e2e-verify.common.sh#L196

But slsa-verifier will pass the verification if ANY of the Provenance's Envelope signatures is good.
 - https://github.com/slsa-framework/slsa-verifier/blame/d96b9777090694fa5096ee1b9c710a46b5a66f5e/verifiers/internal/gcb/provenance.go#L617-L636

To fix, I edit the script to change the Payload for all of the embedded Provenance Envelopes.

## Testing

Executed the `e2e.gcb.push.main.default.slsa3.yml` workflow against this PR's branch, and it did resolve the issue.
 - https://github.com/slsa-framework/example-package/actions/runs/10066797611/job/27828961436#step:5:150
 - https://github.com/slsa-framework/slsa-github-generator/issues/3344#issuecomment-2246368843

### CODE REVIEW

**Comment by ramonpetgrave64** [comment]
@laurentsimon

File: .github/workflows/scripts/e2e-verify.common.sh [hunk 1]
```diff
@@ -193,7 +193,7 @@ e2e_set_payload() {
     local this_builder
     this_builder=$(e2e_this_builder)
     if [[ "${this_builder}" == "gcb" ]]; then
-        jq -c ".provenance_summary.provenance[0].envelope.payload = \"$(echo "$2" | base64 -w0)\"" <"$1"
+        jq -c ".provenance_summary.provenance[0].envelope.payload = \"$(echo "$2" | base64 -w0)\"" <"$1" > tmp.json && mv tmp.json "$1"
```
Comment (laurentsimon) [inline .github/workflows/scripts/e2e-verify.common.sh:196]: why do we need to create a file? I think the function needs to print the change, and the caller saves it to file of they want?

Comment (ramonpetgrave64) [reply .github/workflows/scripts/e2e-verify.common.sh:196]: ok. something isn't happening correctly here for gcb, then
https://github.com/slsa-framework/example-package/blob/441e65668f26e0018b31cc31955d6d8264304692/.github/workflows/scripts/e2e-verify.common.sh#L540-L546

Like `$bad_provenance` should actually be `$PROVENANCE`

**Comment by ramonpetgrave64** [comment]
@haydentherapper @laurentsimon

**Review by haydentherapper — APPROVED** [review]
Clean fix, thanks!

**Pull request merged**